rfcs/0000-windows-containers.md

```diff
@@ -0,0 +1,73 @@
+## Problem:
+
+kpack does not support building containers that can run on windows
+
+## Outcomes:
+
+### kpack should run on clusters that have windows workers
+
+kpack service pods and build pods for linux images will need to know not to run on the windows workers to work in a mixed-os cluster.
+
+The kpack release configuration should specify to run pods explicitly on linux os.
+The kpack controller should schedule linux builds on linux workers.
+
+**Actions to Take**
+
+Use the following node selector for all service pods and build pods for linux images:
+
+```
+nodeSelector:
+    kubernetes.io/os: linux
+```
+
+### kpack should support creating windows builders
+
+This is currently supported in the [pack cli](https://github.com/buildpacks/pack/issues/469)
+
+**Actions to Take**
+
+- kpack-controller will use a new environment variable `WINDOWS_LIFECYCLE_IMAGE` to reference the windows lifecycle image
+- When a Builder or ClusterBuilder is created using a windows stack and windows buildpackages, kpack-controller must create a windows builder using the windows lifecycle image.
+    - Image os is determined by the image [os property](https://github.com/opencontainers/image-spec/blob/master/config.md#:~:text=os%20string).
+
+Notes:
+
+- Builders with mixed os images must error gracefully
+- Builders should be labeled with `os.kpack.io: windows` for os disambiguation independent of image metadata
```

Comment:
Why do we need to label kpack produced builder images with our own os label?

Reply:
I think this removes the need to talk to the registry to determine the os

Reply:
At what point in the kpack lifecycle?

Reply:
creating builds time

Reply:
Is this label a k8s label or a registry label? We already fetch the registry on each build pod creation.

Reply:
k8s

in all of the cases I've added a `os.kpack.io: windows`, if we already read the image then it's not necessary

Reply:
We could add an `os` key to the builder/image/stack/(store?) status but, It does not seem necessary and probably could be skipped in the rfc.

Reply:
the resource label is now not a part of the rfc